### PR TITLE
Add CRUD

### PR DESCRIPTION
**Pull request by jxb345**
- Added 'createOne' function to create a single item.
- Added post, delete, and put function to Descriptions component
- Added three endpoints ('/post', '/delete', and '/put' to server/index.js
- Added saveOne, deleteOne, updateOne functions to database/index.js

### CODE REVIEW

File: client/index.jsx [hunk 1]
```diff
@@ -86,6 +90,46 @@ class Descriptions extends React.Component {
     });
   }
 
+  handlePost() {
```
Comment (steven-miller) [inline client/index.jsx]: This is pretty handy, but doubt we need to add anything to the front-end - the endpoints should work without adding them to React

File: database/createOne.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+const faker = require('faker');
+const generator = require('../helpers/generator.js');
+
+const item = () => {
```
Comment (steven-miller) [inline database/createOne.js]: We could ask, but it feels like this should receive args from the endpoints to let you pass in your own creation info. i.e. if I `POST` I want to decide what I'm creating, not just that I'm creating a thing

File: server/index.js [hunk 3]
```diff
@@ -30,6 +36,34 @@ app.get('/description', (req, res) => {
...
+app.post('/post', (req, res) => {
+  const post = req.body.postItem[0];
+  // req.setTimeout(0);
+  db.saveOne(post, (err) => {
+    if (err) { throw err; }
+    res.status(200);
+  });
+});
+
+app.delete('/delete', (req, res) => {
```
Comment (steven-miller) [inline server/index.js]: This is just typical REST API nomenclature, but you'll probably want your delete call to have the ID of the item present in the URL. E.g. `/delete/:id`, where `:id` is the ID of the item in the database.

Check out e.g. the IMGUR API where you delete in this same way: https://api.imgur.com/endpoints/image/#image-delete

File: client/index.jsx (hunk 1 above)
Comment (jxb345) [reply client/index.jsx]: Yes, I know I asked about this last week and your feeling was that we didn't need to touch the front-end, but this was my only idea for implementing CRUD, so I forged ahead.  You're probably right, but I'd like to get some clarification, which I'll do.

File: database/createOne.js (hunk 2 above)
Comment (jxb345) [reply database/createOne.js]: Your reasoning makes sense. I was thinking that the critical part of the POST was the request being sent from the client to the server rather who determines the actual content of the POST (user versus the code).

File: server/index.js (hunk 3 above)
Comment (jxb345) [reply server/index.js]: Yeah, your suggestion fits a more "real-world" scenario. I'll make that change.

**Pull request merged**